File: modules/binary/app/model/NumberBase.java
package model;

public enum NumberBase {

  // @formatter:off
  BINARY      (2,   "Binär",        "0b", "0-1",          "0-1"),
  OCTAL       (8,   "Oktal",        "0o", "1-7",        "0-7"),
  HEXADECIMAL (16,  "Hexadezimal",  "0x", "1-9a-fA-F",  "0-9a-fA-F"),
  DECIMAL     (10,  "Dezimal",      "",   "1-9",        "0-9");
  //@formatter:on

  private int base;

  private String systemName;
  private String mark;
  private String regexStart;
  private String regexRest;

  private NumberBase(int theBase, String theBaseName, String theMark, String theRegexStart, String theRegexRest) {
    base = theBase;
    systemName = theBaseName;
    mark = theMark;
    regexStart = theRegexStart;
    regexRest = theRegexRest;
  }

  public int getBase() {
    return base;
  }

  public String getHtmlPattern() {
    return "[\\s" + regexStart + "][\\s" + regexRest + "]*";
  }

  public String getMark() {
    return mark;
  }

  public String getPluralName() {
    return systemName + "zahlen";
  }

  public String getRegex() {
    return "-?" + mark + "[" + regexStart + "][" + regexRest + "]*";
  }

  public String getSingularName() {
    return systemName + "zahl";
  }

  public String getSystemName() {
    return systemName + "system";
  }

}

File: server/app/model/tools/nary/NumberBase.scala
package model.tools.nary

import enumeratum.{EnumEntry, PlayEnum}
import play.twirl.api.Html

import scala.collection.immutable

sealed abstract class NumberBase(val base: Int, val baseName: String, val mark: String, val regexStart: String, val regexRest: String) extends EnumEntry {

  val htmlPattern: String = "[\\s" + regexRest + "]+"

  val pluralName: String = baseName + "zahlen"

  val regex: String = "-?" + mark + "[" + regexStart + "][" + regexRest + "]*"

  val singularName: String = baseName + "zahl"

  val systemName: String = baseName + "system"

}

object NumberBase extends PlayEnum[NumberBase] {

  override val values: immutable.IndexedSeq[NumberBase] = findValues

  case object Binary extends NumberBase(2, "Binär", "0b", "0-1", "0-1")

  case object Octal extends NumberBase(8, "Oktal", "0o", "1-7", "0-7")

  case object HexaDecimal extends NumberBase(16, "Hexadezimal", "0x", "1-9a-fA-F", "0-9a-fA-F")

  case object Decimal extends NumberBase(10, "Dezimal", "", "1-9", "0-9")

}
